### PR TITLE
ci(coverage): Set the correct out format for CI coverage reports

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,3 +11,6 @@ lto = true
 # Optimize quote even in debug mode. Speeds up proc-macros enough to account
 # for the extra time of optimizing it for a clean build of matrix-sdk-ffi.
 quote = { opt-level = 2 }
+# Optimize pbkdf2 in debug mode. Speeds up tests that derive encryption keys
+# using pbkdf2, mainly the store integration tests.
+pbkdf2 = { opt-level = 2 }
```
Comment (jplatte) [inline Cargo.toml:16]: Should be a separate commit.

**Review by jplatte — APPROVED** [review]

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ jobs:
       uses: actions-rs/cargo@v1
       with:
         command: tarpaulin
-        args: --workspace
+        args: --out Xml
```
Comment (Hywan) [inline .github/workflows/coverage.yml:41]: It's my fault, I've removed it by accident in https://github.com/matrix-org/matrix-rust-sdk/pull/753. Sorry for that.

Comment (poljar) [reply .github/workflows/coverage.yml:41]: It happens, also I approved the PR... Besides, we're still not sure that this fixes the problem.

Comment (poljar) [reply .github/workflows/coverage.yml:41]: Seems to be fixed indeed.
